### ISSUE TITLE
Nice-to-have Features

### ISSUE THREAD

**Issue by rodrihgh** [dieghernan/RKI-Corona-Atlas#2]
@dieghernan 

- [x] Redraw risk areas while dragging (actual behaviour seemingly linked to `mouseup` ). [Leaflet reference](https://leafletjs.com/reference-1.7.1.html#map-zoomlevelschange)
    - [x] Alternatively/complementary, paint the off-chart areas of on-chart regions to avoid glitchy square cutoffs while dragging.

- [x] Add status-change dates on the tables and the clickable information on the map.
    - [x] German
    - [x] English

**Comment by dieghernan** [comment]
Regarding the first issue, don't know how to do it even in pure Leaflet (or even possible, [the official docs](https://leafletjs.com/examples/choropleth/) shows the same glitch). 

If solved, it should be somehow implemented on R (note that R uses https://leafletjs.com/reference-1.3.4.html, not 1.7.1).

Other alternative could be using points instead of polygons: http://rstudio.github.io/leaflet/markers.html#marker-clusters

Last option would be not using interactivity and creating static maps, although personally I would prefer to have interactive maps.

See an example of a full html Leaflet map (create an empty test.html file and copy the code inside):

```html
<!DOCTYPE html>
<html>
<head>
	<title>Minimal page | leaflet-providersESP</title>
	<meta charset="utf-8" />
	<meta name="viewport" content="width=device-width, initial-scale=1.0">
	<!-- Load Leaflet -->
	<link rel="stylesheet" href="https://unpkg.com/leaflet@1.7.1/dist/leaflet.css" />
	<script src="https://unpkg.com/leaflet@1.7.1/dist/leaflet.js"></script>
	<script src="https://cdn.jsdelivr.net/gh/calvinmetcalf/leaflet-ajax/dist/leaflet.ajax.min.js"></script>
	<!-- Display map full page -->
	<style>
	html {
		height: 100%
	}
	body {
		height: 100%;
		margin: 0;
		padding: 0;
	}
	#mapid {
		height: 100%;
		width: 100%
	}
	</style>
</head>
<body>
	<!-- Create map -->
	<div id="mapid"></div>
	<script>
	var mymap = L.map('mapid').setView([51.705533,  11.8124408], 4);
	
	L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
    attribution: '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
}).addTo(mymap);
	var geojsonLayer = new L.GeoJSON.AJAX("https://gisco-services.ec.europa.eu/distribution/v2/countries/geojson/CNTR_BN_60M_2020_4326.geojson");       
    geojsonLayer.addTo(mymap);

	
	</script>
</body>

```

**Comment by rodrihgh** [comment]
The first issue can be alleviated on Javascript with the option `{renderer: L.svg({ padding: 0.5 })}`. The padding value can be tuned to find a good performance trade-off. See Leaflet/Leaflet#2814.

@dieghernan can you add the option directly on the R code?

```html
<!DOCTYPE html>
<html>
<head>
	<title>Minimal page | leaflet-providersESP</title>
	<meta charset="utf-8" />
	<meta name="viewport" content="width=device-width, initial-scale=1.0">
	<!-- Load Leaflet -->
	<link rel="stylesheet" href="https://unpkg.com/leaflet@1.7.1/dist/leaflet.css" />
	<script src="https://unpkg.com/leaflet@1.7.1/dist/leaflet.js"></script>
	<script src="https://cdn.jsdelivr.net/gh/calvinmetcalf/leaflet-ajax/dist/leaflet.ajax.min.js"></script>
	<!-- Display map full page -->
	<style>
	html {
		height: 100%
	}
	body {
		height: 100%;
		margin: 0;
		padding: 0;
	}
	#mapid {
		height: 100%;
		width: 100%
	}
	</style>
</head>
<body>
	<!-- Create map -->
	<div id="mapid"></div>
	<script>
	var mymap = L.map('mapid', {renderer: L.svg({ padding: 0.5 })}).setView([51.705533,  11.8124408], 4);

	L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
    attribution: '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
}).addTo(mymap);
	var geojsonLayer = new L.GeoJSON.AJAX("https://gisco-services.ec.europa.eu/distribution/v2/countries/geojson/CNTR_BN_60M_2020_4326.geojson");
    geojsonLayer.addTo(mymap);


	</script>
</body>
```

**Comment by dieghernan** [comment]
Nice!! I have been investigating how to implement this on R Leaflet with no luck, so I would focus on other tasks by now.

**Comment by dieghernan** [comment]
Regarding status-change dates, shouldn’t be a problem as long as we get the data. 

By now, information is retrieved manually, so it would be a matter of adding a new column on db_countries.csv. Note that this feature would require more man-power, as currently there is not a safe way for scrapping data from the pdfs.

**Comment by dieghernan** [comment]
@rodrihgh meet the magic code line

https://github.com/dieghernan/RKI-Corona-Atlas/blob/956c284ebf080c92cd21a6df5acdf4b9318fd94b/_R/plugins/leaflet-1.3.1/leaflet-mod.js#L11744

**Comment by rodrihgh** [comment]
@dieghernan check out https://github.com/dieghernan/RKI-Corona-Atlas/commit/a34c695eea726decf47b9ff4fb3d96f4d6c4c6d5

I have finished a somewhat stable scrapper off the German website. On top of the basic info we already had, it extracts the German comments for every country. I have also parsed the starting date of the current status in the column `risk_date`. Check out the resulting [db_scrapped.csv](https://github.com/dieghernan/RKI-Corona-Atlas/blob/dev/assets/data/db_scrapped.csv)

I would rather not scrap the English PDF for the time being, since PDF scrapping looks too daunting to me in comparison with HTML. But there is the German info and the `risk_date`, so next week we can talk about adding extra information in the map :)

**Comment by dieghernan** [comment]
All that is now implemented after #3

**Comment by dieghernan** [comment]
Reopened to track new features

**Comment by rodrihgh** [comment]
For the info in the English map I would do something simple like:

**NAME_ENGL**
Risk designation
`% if country.risk_date %`
since `country.risk_date`
`% endif %`

`% if country.regions.size > 0 %`
A different risk level may apply for the following regions:
`% for region in country.regions %`

- `region.NAME_ENGL`, since `region.risk_date`

`% endfor %`

**Comment by dieghernan** [comment]
> For the info in the English map I would do something simple like:
> 
> **NAME_ENGL**
> Risk designation
> `% if country.risk_date %`
> since `country.risk_date`
> `% endif %`
> 
> `% if country.regions.size > 0 %`
> A different risk level may apply for the following regions:
> `% for region in country.regions %`
> 
> * `region.NAME_ENGL`, since `region.risk_date`
> 
> `% endfor %`

 See https://github.com/dieghernan/RKI-Corona-Atlas/commit/cd6bb0ab928c0e0635b1646e4fae72e70eb08f06
